src/Error.h

```diff
@@ -92,13 +94,31 @@ struct ErrorReport {
      * this by only actually throwing if there isn't an exception in
      * flight already.
      */
-#if __cplusplus >= 201100 || _MSC_VER >= 1900
-    ~ErrorReport() noexcept(false);
+#if (__cplusplus >= 201100 || _MSC_VER >= 1900) && defined(WITH_EXCEPTIONS)
+    [[ noreturn ]] ~ErrorReport() noexcept(false);
 #else
-    ~ErrorReport();
+    [[ noreturn ]] ~ErrorReport();
```

Comment:
If I'm reading it right, this says:

if C++11 and WITH_EXCEPTIONS
   use C++11 thing and exception thing
else
   use C++11 thing
endif

So in the case where C++11 is off, we're falling back to using a C++11 construct. But we require C++11 now, so maybe we should just remove the C++11 clause of the if?

Reply:
Heh, yeah, good point; that clause must be there for quite a while. I'll double-check the edit history but will likely remove the C++11 check.

Reply:
...it wasn't even the correct check for C++11; canonically, it should be `__cplusplus >= 201103`